File: src/weaponinventorymod/gui/WimGuiButtonColors.java
package weaponinventorymod.gui;

import java.awt.Color;

final class WimGuiButtonColors {
    final Color idle;
    final Color hover;

    WimGuiButtonColors(Color idle, Color hover) {
        this.idle = idle;
        this.hover = hover == null ? idle : hover;
    }

    static WimGuiButtonColors same(Color color) {
        return new WimGuiButtonColors(color, color);
    }
}

File: src/main/kotlin/weaponsprocurement/ui/WimGuiButtonColors.kt
package weaponsprocurement.ui

import java.awt.Color

class WimGuiButtonColors(
    @JvmField val idle: Color,
    hover: Color?,
) {
    @JvmField
    val hover: Color = hover ?: idle

    companion object {
        @JvmStatic
        fun dimmedInner(color: Color): WimGuiButtonColors = WimGuiButtonColors(color, color)

        @JvmStatic
        fun same(color: Color): WimGuiButtonColors = dimmedInner(color)
    }
}
